requirements.txt

```diff
@@ -1,5 +1,7 @@
 black==21.7b0
 mypy==0.931
+git+https://github.com/algorand/py-algorand-sdk@teal-blackbox
 pytest
+pytest-cov
 pytest-timeout
-py-algorand-sdk
+tabulate==0.8.9
```

Comment:
@tzaffi For my understanding:  It seems pyteal does _not_ use _tablulate_.  Though I know py-algorand-sdk uses the dependency.  I'm wondering if the dependency is declared to explicitly state the transitive dependency or if there's another reason?

I'm asking the question as a way to learn a bit more about pip conventions/norms.

Reply:
I believe this is a temporary artifact of me `pip install`ing our **PySDK** from github instead of from PyPI, so `pip` isn't smart enough to compute the transitive dependency and I had to add it here. But once my PR gets merged in the PySDK repo, we should double check it.

Thinking more along these lines, I realized that there is a more general problem? How do you tell that all of our requirements are actually necessary?

Here's a possible way:

1. We should add autoflake (as you suggested) to our CI and use it to enforce having no unused imports
2. We should then be able to use `pipreqs` to [validate that all the items](https://python-bloggers.com/2021/10/how-to-generate-the-requirements-of-your-python-project-based-on-your-imports/) in `requirements.txt` are actually needed

Ironically, it's exactly this situation that would be a false positive...

Reply:
@tzaffi Got it - thanks for the response.

* Considering the situation more broadly, I've also wondered how we'd separate src vs test dependencies.  I imagine that requires crossing into a more sophisticated build tool.  
* _pipreqs_ looks interesting.  Thanks for raising.  I added a note into https://github.com/algorand/pyteal/issues/218.
* If it's OK with you, let's leave the thread open until py-algorand-sdk is merged?  It can serve as a visual reminder to double-check the dependencies.

Reply:
👍

Reply:
Now we need `tabulate` because of `graviton`